File: python/consumer.py
# https://github.com/thanhson1085/python-kafka-avro
from kafka import KafkaConsumer
import avro.schema
import avro.io
import io
import datetime

# Kafka settings
kafkaHost = "localhost:9092"
kafkaTopic = "avro"

consumer = KafkaConsumer(kafkaTopic, bootstrap_servers=[kafkaHost])

schema_path="../schemas/user.avsc"
schema = avro.schema.parse(open(schema_path).read())

for msg in consumer:
    bytes_reader = io.BytesIO(msg.value)
    decoder = avro.io.BinaryDecoder(bytes_reader)
    reader = avro.io.DatumReader(schema)
    decoded_message = reader.read(decoder)
    print("Message processed at {}".format(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')))
    print(decoded_message)

File: ruby/consumer.rb
require "avro"
require "kafka"

KAFKA_HOST = "localhost:9092"
KAFKA_TOPIC = "avro"

schema = Avro::Schema.parse(File.open("../schemas/user.avsc", "rb").read)

kafka = Kafka.new(seed_brokers: [KAFKA_HOST])
consumer = kafka.consumer(group_id: "avro-test")
consumer.subscribe(KAFKA_TOPIC, start_from_beginning: false)

consumer.each_message do |message|
  begin
    buffer = StringIO.new(message.value)
    decoder = Avro::IO::BinaryDecoder.new(buffer)
    datumreader = Avro::IO::DatumReader.new(schema)
    decoded_message = datumreader.read(decoder)
    puts "Message processed at #{Time.now.strftime("%Y-%m-%d %H:%M")}"
    puts decoded_message
  rescue
    puts "Can't process message"
  end
end
